### DISCUSSION TITLE
opengothic-v1.0.1878

### DISCUSSION THREAD

**Discussion by Try** [Try/OpenGothic#389] [General]
![изображение](https://user-images.githubusercontent.com/122894/213894755-bcd7954c-fe59-4e72-a27c-e04df59f237c.png)

* Game
  * Improve path-finding algorithm, for AI
  * Adjust combo-attacks and battle animations to vanilla
  * Added specific controls, related to Gothic2 control scheme
  * Fix sneaking with bow animation
  * Added support for ladders (#325)

* General
  * Fix camera stutters
  * Fix behavior of -rt command line key
  * Added marin 'set time' command
  * Add support for hideFocus and discard noFrate options in *.ini file (#364)
  * Allow to pick items and use bow, while sneaking

* Graphics
  * Switch rendering engine to linear color space
  * Optimize SSAO, with help of compute shaders 
  * Improve sky scattering
  * Added ray-query shadows
  * Migrate geometry rendering to GL_EXT_mesh_shader
  * Added per-particle trails 
  * 32-bit depth buffer is now default
  * Added support for soft-shadows
  
* Music
  * Update OpenAL to 1.22.2
  * Enable music on MacOS
  * Fix loading of sound variations (known as sheep sounds problem)
  
* MacOS
  * Improve full-screen mode on Mac
  
* Bugs
  * Avoid false-locking of UI thread, while loading
  * Fix crash in npc_getequippedrangedweapon
  * Fix exception handler in TriggerScript
  * Fix ai_printscreen execution order
  * Fix issue, that caused invalid world-state restore in some mods

* UPD (Hotfix 1.0.1891)
  * Bugfix for OpenGothic to Union interaction
  * Bugfix for Serialization issues on Linux

  

<hr /><em>This discussion was created from the release <a href='https://github.com/Try/OpenGothic/releases/tag/opengothic-v1.0.1891'>opengothic-v1.0.1891</a>.</em>

**Comment by JanSimek** [comment]
With this release I don't have any sound problems on my Arch Linux anymore! https://github.com/Try/OpenGothic/issues/342

**Comment by TomislavTom** [comment]
hi, can you make a little showcase video and explain what is still missing? congratulations and keep up the awesome work!

**Reply by Try** [reply]
Hi, @TheDovahkiin97 !

I remember, last time we discuss this in context of graphics. As usual it's a bottomless topic, yet before making video there is 2 features I would like to add:
1. Stochastic [path-racer](https://blog.demofox.org/2020/05/10/ray-marching-fog-with-blue-noise/) for atmosphere
2. Better water shader (not 100% sure, what it's going to be)

As for game-play - should work very close to vanilla already

**Reply by TomislavTom** [reply]
Nice. Do you have an idea how long it might take to implement those? Roughly.

I will try all of this out in approximately one month.
Will it be a true release with promotions?

Do you think that OpenGothic already is the best way to play and enjoy Gothic?

**Reply by Try** [reply]
> Will it be a true release with promotions?

Actually, I haven't thought of that. What exactly do you mean by promotions? Like youtube stuff + reddit + else?

> Do you think that OpenGothic already is the best way to play and enjoy Gothic?

It would be very bold to claim so, but one of the best for sure ;)

**Reply by TomislavTom** [reply]
Yes I meant exactly those platforms.

What would be the alternatives? I can only think of the dx11 renderer.
Almost everything else worth mentioning should be compatible with your mod. Like l'hiver. Right?

**Reply by Try** [reply]
Sound like something similar to this in style:
https://www.youtube.com/watch?v=EXdxeJs4apU

**Comment by unknown-kmt** [comment]
Hello. Is there any chance to have a port for switch homebrew?

**Reply by Try** [reply]
Hi, not on my side in any case.
SteamDeck is already out there and AFAIK OpenGothic works as-is there with no issues. Meanwhile Switch doesn't even have full vulkan support, what means constant baby-sitting with this platform
